java-checks/src/main/java/org/sonar/java/checks/naming/MethodNamedHashcodeOrEqualCheck.java

```diff
@@ -37,16 +37,23 @@ public List<Tree.Kind> nodesToVisit() {
 
   @Override
   public void visitNode(Tree tree) {
-    IdentifierTree methodIdentifier = ((MethodTree) tree).simpleName();
+    MethodTree method = (MethodTree) tree;
+    IdentifierTree methodIdentifier = method.simpleName();
     switch (methodIdentifier.name()) {
       case "hashcode":
-        report(methodIdentifier, "hashCode()");
+        if (notOverriding(method)) {
+          report(methodIdentifier, "hashCode()");
```

Comment:
To not duplicate 3 times `notOverriding()` in 3 distincts places, what about renaming `report()` into `reportIfNotOverriding()` and do like this:
```
case "hashcode":
  reportIfNotOverriding(method, "hashCode()");
```

and later replace:
```
private void report(IdentifierTree methodIdentifier, String substitute) {
...
```
by
```
private void reportIfNotOverriding(MethodTree method, String substitute) {
  if (notOverriding(method)) {
    reportIssue(method.simpleName(), "Either override Object." + substitute + ", or totally rename the method to prevent any confusion.");
  }
}
```